pkg/osquery/runtime/runner.go

```diff
@@ -27,14 +28,16 @@ type settingsStoreWriter interface {
 
 type Runner struct {
 	registrationIds []string                    // we expect to run one instance per registration ID
+	regIDLock       sync.Mutex                  // locks access to registrationIds
```

Comment:
I discovered that if we are locking for a single value, there is an `atmoic.Value` or `atmoic.Pointer` that can be used, like:

```go
	var registrationIds atomic.Value
	registrationIds.Store([]string{"a", "b", "c"})
	theIds := registrationIds.Load().([]string)
```
 
or

```go
	var registrationIds atomic.Pointer[[]string]
	registrationIds.Store(&[]string{"a", "b", "c"})
	theIds := registrationIds.Load()
```

the internets generally say that atmoics are faster ... but I doubt that it makes any noticeable difference for us, also pointer to an array feels weird and I'm not sure of the implications

feel free to not act on this, I don't know if it's any better